framework/fp_vectors/Floats.h

```diff
@@ -30,34 +34,72 @@ typedef __fp16 fp16_t;
 #endif
 
 #define FLOAT16_EXPONENT_MASK  0x1fu
+#define BFLOAT16_EXPONENT_MASK 0xffu
 #define FLOAT32_EXPONENT_MASK  0xffu
 #define FLOAT64_EXPONENT_MASK  0x7ffu
 #define FLOAT80_EXPONENT_MASK  0x7fffu
 
 #define FLOAT16_INFINITY_EXPONENT  0x1fu
+#define BFLOAT16_INFINITY_EXPONENT 0xffu
 #define FLOAT32_INFINITY_EXPONENT  0xffu
 #define FLOAT64_INFINITY_EXPONENT  0x7ffu
 #define FLOAT80_INFINITY_EXPONENT  0x7fffu
 
 #define FLOAT16_NAN_EXPONENT  0x1fu
+#define BFLOAT16_NAN_EXPONENT 0xffu
 #define FLOAT32_NAN_EXPONENT  0xffu
 #define FLOAT64_NAN_EXPONENT  0x7ffu
 #define FLOAT80_NAN_EXPONENT  0x7fffu
 
+#define FLOAT16_DENORM_EXPONENT  0x00
+#define BFLOAT16_DENORM_EXPONENT 0x00
+#define FLOAT32_DENORM_EXPONENT  0x00
+#define FLOAT64_DENORM_EXPONENT  0x00
+#define FLOAT80_DENORM_EXPONENT  0x00
+
 #define FLOAT16_EXPONENT_BIAS  0x0fu
+#define BFLOAT16_EXPONENT_BIAS 0x7fu
 #define FLOAT32_EXPONENT_BIAS  0x7fu
 #define FLOAT64_EXPONENT_BIAS  0x3ffu
 #define FLOAT80_EXPONENT_BIAS  0x3fffu
 
 #define FLOAT16_MANTISSA_MASK  0x3ffu
+#define BFLOAT16_MANTISSA_MASK 0x7fu
 #define FLOAT32_MANTISSA_MASK  0x7fffffu
 #define FLOAT64_MANTISSA_MASK  0xfffffffffffffu
-#define FLOAT80_MANTISSA_MASK  0xffffffffffffffffu
+#define FLOAT80_MANTISSA_MASK  0x7fffffffffffffffu
 
 #define FLOAT16_MANTISSA_QUIET_NAN_MASK  0x200u
+#define BFLOAT16_MANTISSA_QUIET_NAN_MASK 0x40u
 #define FLOAT32_MANTISSA_QUIET_NAN_MASK  0x400000u
 #define FLOAT64_MANTISSA_QUIET_NAN_MASK  0x8000000000000u
-#define FLOAT80_MANTISSA_QUIET_NAN_MASK  0x8000000000000000u
+#define FLOAT80_MANTISSA_QUIET_NAN_MASK  0x4000000000000000u
+
+#define FP16_SIGN_BITS        1
+#define FP16_EXPONENT_BITS    5
+#define FP16_MANTISSA_BITS    10
+#define FP16_QUIET_BITS       1
+
+#define BFLT16_SIGN_BITS      1
+#define BFLT16_EXPONENT_BITS  8
+#define BFLT16_MANTISSA_BITS  7
+#define BFLT16_QUIET_BITS     1
```

Comment:
Can we choose one of "FLT" or "FLOAT" only? Clean ups of existing API can be a new PR.